src/server/dflycmd.h

```diff
@@ -232,6 +247,11 @@ class DflyCmd {
   using ReplicaInfoMap = absl::btree_map<uint32_t, std::shared_ptr<ReplicaInfo>>;
   ReplicaInfoMap replica_infos_ ABSL_GUARDED_BY(mu_);
 
+  // We only acquire a write lock in TakeOver. This happens only once, as the state
+  // after the takeover is TAKEN_OVER. All other paths, acquire it in read only mode.
+  util::fb2::SharedMutex last_takeover_src_mu_;
+  std::shared_ptr<ReplicaInfo> last_takeover_source_;
```

Comment:
TODO separate PR.

I spoke with @andydunstall and we would also like to bring back and taken over node via `REPLICAOF` command. 

So the flow is:

1. node a master, node b replica
2. node b take over a (a is TAKEN_OVER state. b is master now)
3. after some time connect to node a, and `REPLICAOF B` 

As is `REPLICAOF` would fail because of the state. 

We should make the necessary changes to support that